File: src/obstacle/Obstacle.cpp
// Copyright (c) TrajoptLib contributors

#include "obstacle/Obstacle.h"

#include <vector>

#include "obstacle/ObstaclePoint.h"

namespace trajopt {

Obstacle::Obstacle(double safetyDistance, std::vector<ObstaclePoint> points)
    : safetyDistance(safetyDistance), points(std::move(points)) {}

}  // namespace trajopt

File: java/src/main/java/org/sleipnirgroup/trajopt/Obstacle.java
// Copyright (c) TrajoptLib contributors

package org.sleipnirgroup.trajopt;

import static org.sleipnirgroup.util.ObjectChecker.requireNonNullAndWrapUnmodifiable;

import java.util.List;

public final class Obstacle {
    public final double safetyDistance;
    public final List<? extends ObstaclePoint> points;

    public Obstacle(double safetyDistance, boolean applyToAllSegments,
            List<? extends ObstaclePoint> points) throws NullPointerException {
        this.safetyDistance = safetyDistance;
        this.points = requireNonNullAndWrapUnmodifiable(points, "List of obstacle points cannot be null", "Obstacle point cannot be null");
    }

    @Override
    public String toString() {
        return "{\"safety_distance\": " + safetyDistance
            + ", \"points\": " + points
            + "}";
    }
}
